File: applications/bank-account-sink/src/main/java/com/vmware/bank/account/sink/GemFireConfig.java
package com.vmware.bank.account.sink;

import com.vmware.financial.open.banking.account.service.AccountService;
import com.vmware.financial.open.banking.domain.account.BankAccount;
import com.vmware.financial.open.banking.gemfire.account.event.service.AccountDataService;
import org.apache.geode.cache.DataPolicy;
import org.apache.geode.cache.GemFireCache;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import org.springframework.data.gemfire.GemfireTemplate;
import org.springframework.data.gemfire.client.ClientRegionFactoryBean;
import org.springframework.data.gemfire.config.annotation.ClientCacheApplication;
import org.springframework.data.gemfire.repository.config.EnableGemfireRepositories;


@Configuration
@Profile("gemfire")
@ClientCacheApplication
@EnableGemfireRepositories
public class GemFireConfig {
    public GemFireConfig()
    {
        System.out.printf("created");
    }

    //create region --name=BankAccount --type=PARTITION
    @Bean
    ClientRegionFactoryBean<String, BankAccount> bankAccount(GemFireCache gemFireCache)
    {
        var factory = new ClientRegionFactoryBean<String, BankAccount>();
        factory.setCache(gemFireCache);
        factory.setName("BankAccount");
        factory.setDataPolicy(DataPolicy.EMPTY);
        return factory;
    }

    @Bean
    AccountService dataService(GemFireCache gemFireCache, GemfireTemplate gemfireTemplate)
    {
        return new AccountDataService(gemfireTemplate);
    }
}

File: applications/bank-account-rest-service/src/main/kotlin/com/vmware/financial/open/banking/account/GemFireConfig.kt
package com.vmware.financial.open.banking.account

import com.vmware.financial.open.banking.account.repository.AccountRepository
import com.vmware.financial.open.banking.account.service.AccountDataService
import com.vmware.financial.open.banking.domain.account.BankAccount
import org.apache.geode.cache.DataPolicy
import org.apache.geode.cache.GemFireCache
import org.apache.geode.pdx.ReflectionBasedAutoSerializer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.data.gemfire.GemfireTemplate
import org.springframework.data.gemfire.client.ClientRegionFactoryBean
import org.springframework.data.gemfire.config.annotation.ClientCacheApplication
import org.springframework.data.gemfire.config.annotation.EnableContinuousQueries
import org.springframework.data.gemfire.config.annotation.EnablePdx
import org.springframework.data.gemfire.repository.config.EnableGemfireRepositories

/**
 * @author Gregory Green
 */
@Configuration
@EnableContinuousQueries
@EnableGemfireRepositories(basePackageClasses = [AccountRepository::class])
@ClientCacheApplication(subscriptionEnabled = true)
@EnablePdx
class GemFireConfig {

    @Bean
    fun serializer() : ReflectionBasedAutoSerializer {
        return ReflectionBasedAutoSerializer(".*")
    }
    @Bean("bankAccountRegion")
    fun bankAccountRegion(gemFireCache: GemFireCache) : ClientRegionFactoryBean<String,BankAccount>
    {
        var bankAccountRegion = ClientRegionFactoryBean<String, BankAccount>()
        bankAccountRegion.cache = gemFireCache
        bankAccountRegion.setRegionName("BankAccount")
        bankAccountRegion.setDataPolicy(DataPolicy.EMPTY)
        return bankAccountRegion;
    }

    @Bean
    fun gemfireTemple(gemFireCache: GemFireCache, region: ClientRegionFactoryBean<String,BankAccount>) : GemfireTemplate{
        return GemfireTemplate(region.region);
    }
}